Allow to set the maximum number of transfers in the UI

schaerfo opened the pull request:
This is how it looks:

![grafik](https://github.com/user-attachments/assets/68a6fef0-f219-4fca-b586-4e4413e1b027)

Feel free to make changes, I am not very familiar with CSS/Tailwind

MichaelKutzner commented:
Thank you for bringing this to my attention.
For isochrones, the upper limit is set to match the server limit. As the number isn't that high, I found a drop-down menu still easy to use.
Right now, this option is not enabled on the “Connection” tab. But this can be changes without much afford.
The current design looks like that:
![image](https://github.com/user-attachments/assets/bf218561-76b8-4822-9e4e-9371f20223e1)

schaerfo commented:
Great, I'll take a look at the `NumberSelect` you merged and update this PR

schaerfo commented:
Ok, I rebased the patch and it got a quite a bit smaller :) It now looks like this:
<img width="544" height="621" alt="grafik" src="https://github.com/user-attachments/assets/e8a047fa-e08b-48ce-b772-20f81d66ead9" />
* The connections and isochrones tabs share state, so setting max transfers in one tab will also change it in the other
* I removed the switch to enable limiting the number of transfers

traines-source commented:
Nice, what do you think, what is less confusing? Always showing the actual technical upper limit of 14 transfers or showing some special label like "Unlimited" for that in the dropdown which is technically wrong?

felixguendling commented:
> some special label like "Unlimited"

I think the concrete number is better. Maybe even configurable in the server because for small datasets a smaller number might improve performance without any disadvantage for the user.

traines-source commented:
> configurable in the server

Ok but I think this is out of scope for this PR, since we have no straightforward way yet to pass this to the UI.

traines-source commented:
It is, thank you!
